### DISCUSSION TITLE
Node has 4 hops, but graph shows connection to node with only 1 hop?

### DISCUSSION THREAD

**Discussion by stephanvierkant** [zwave-js/zwave-js-ui#3412] [Q&A]
I've changed the location of some of my devices and now my network is terribly slow. One device ('Bureau') even loses connection frequently. I've rebuild the 2network but that didn't make any change.

So I looked at my network graph:

![Screenshot_20231108_092511](https://github.com/zwave-js/zwave-js-ui/assets/601833/6685ed91-bff3-41fc-a8fe-14874096f015)

Blue means 1 hop,  green 2 hops, yellow 3 hops and orange 4 hops. I see 'Lamp' has only one hop, but Bureau has 4. 

How does this work?

**Comment by robertsLando** [comment]
> Blue means 1 hop, green 2 hops, yellow 3 hops and orange 4 hops. I see 'Lamp' has only one hop, but Bureau has 4.

Based on the graph seems Bureau is actually routing like: Bureau --> Lamp --> ??? (the one cropped under lamp) --> Keukenkastje --> Controller.

What I suggest is to change priority route manually to make Bureau to only use Lamp as Repeater so it will have one single hop instead of 3 in order to reach controller. You can do the same if needed with return route

**Reply by stephanvierkant** [reply]
You're right! But why does Bureau (3rd floor) take a detour, while Lamp (3rd floor, next to Bureau) is directly connected to the Controller (1st floor)?

I've tried to set a Priority route, but nothing happens (other than a successfull 'new route set' message)

Now it looks like this:
![Screenshot_20231108-122429.png](https://github.com/zwave-js/zwave-js-ui/assets/601833/e91a9c93-7f57-4b9a-8c59-8abdf46a7d8d)

Lamp, Bureau Anouk and Bureau Stephan are on the 3rd floor, Keukenkastje and Controller are on the first floor. Why doesn't it use any of the many nodes on the 2nd floor? I've got 22 Z-Wave devices well-distributed in my house, so I expect a stronger network.

The only think I can think of: Bureau (2x) and Keukenkastje are all Greenwave nodes, which are not Z-Wave Plus devices.

**Reply by robertsLando** [reply]
> But why does Bureau (3rd floor) take a detour, while Lamp (3rd floor, next to Bureau) is directly connected to the Controller (1st floor)?

Explaining why the zwave network decides to do this is hard, we can only make some guess, mesh is automatically handled by the protocol and sometimes it is known to do some weird thing like this. This is why we added the possibility to customize priority routes so you can override default ones assigned by protocol. Be aware that calling "rebuild routes" against that node will cause it to loose all the priority routes you added

**Reply by robertsLando** [reply]
> I've tried to set a Priority route, but nothing happens (other than a successfull 'new route set' message)

You have set a priorirty RETURN ROUTE, you should set also the other above that (that is the most important one)

![2023-11-08_14-17](https://github.com/zwave-js/zwave-js-ui/assets/11502495/d13c9315-57cb-4128-9e87-9f48765696e7)

For reference: priority route is the route from controller to node, the return route is the route packets follow from node to the controller
